src/OpenTelemetry/Trace/Tracer.cs

```diff
@@ -40,12 +41,10 @@ static Tracer()
         /// </summary>
         /// <param name="spanProcessor">Span processor.</param>
         /// <param name="traceConfig">Trace configuration.</param>
-        public Tracer(SpanProcessor spanProcessor, TraceConfig traceConfig)
+        /// <param name="libraryResource">Resource describing the instrumentation library.</param>
+        public Tracer(SpanProcessor spanProcessor, TraceConfig traceConfig, Resource libraryResource) 
```

Comment:
Could resource be set through TraceConfig to avoid parameters explosion and it should be resource.Empty unless customized)

Reply:
I don't think that's possible, since TraceConfig is (possibly) shared between TracerFactories and this Resource represents the arguments provided to `TracerFactory.GetTracer` (name + version).